### PR TITLE
Adds documentation for the new DownloadClientReadyChecker functionality

### PR DESCRIPTION
**Pull request by zegnus**
## Description of the task

We have a new feature that allows a client to tell the DownloadManager if a download can proceed or not.

This PR adds that information to the public DownloadManagerBuilder

### CODE REVIEW

File: library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java [hunk 1]
```diff
@@ -5,6 +5,11 @@
 
 import com.novoda.downloadmanager.lib.DownloadManager;
 
+/**
+ * A client can specify whether the downloads are allowed to proceed by implementing
+ * {@link com.novoda.downloadmanager.lib.DownloadClientReadyChecker} on your Activity class
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java]: `n your Application class`

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java]: can we stick this in the readme too?

Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java]: maybe, i just think we should be careful to not clutter the readme (even the parralel thing prob shouldn't be in the readme)

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java]: wiki?

**Comment by stefanhoth** [comment]
Sorry, had to be done.

**Pull request merged**